web/client-api/src/main/java/io/deephaven/web/client/api/barrage/WebChunkReaderFactory.java

```diff
@@ -275,7 +275,8 @@ public <T extends WritableChunk<Values>> ChunkReader<T> newReader(
                 final ExpansionKernel<?> kernel =
                         ArrayExpansionKernel.makeExpansionKernel(chunkType, componentTypeInfo.type());
                 final ChunkReader<?> componentReader = newReader(componentTypeInfo, options);
-                return (ChunkReader<T>) new ListChunkReader<>(ListChunkReader.Mode.DENSE, 0, kernel, componentReader);
+                return (ChunkReader<T>) new ListChunkReader<>(ListChunkReader.Mode.VARIABLE, 0, kernel,
```

Comment:
Why hardcoded here?

Reply:
Primary reason is/was that the UI client (at this time) can only speak with a DH server. So, we know that the Schema field is going to be a List. I've added `ListView` and `FixedSizeList` to this case and differentiated the modes.

We don't have a way to currently test the code paths for ListView nor FixedSizeList for the web reader .. but I see that it might be useful if someone finds a way to get their paws on our reader in js or typescript and want to read from a non-deephaven arrow host.

It's also easy for me to add what looks to be correct.